src/Http/Routing/src/RouteEndpointDataSource.cs

```diff
@@ -213,11 +213,11 @@ internal RouteEndpointBuilder GetSingleRouteEndpointBuilder()
             Debug.Assert(entry.InferMetadataFunc != null, "A func to infer metadata must be provided for route handlers.");
 
             rdfOptions = CreateRdfOptions(entry, pattern, builder);
-            rdfMetadataResult = entry.InferMetadataFunc(entry.RouteHandler.Method, rdfOptions);
+            rdfMetadataResult = entry.InferMetadataFunc(methodInfo, rdfOptions);
         }
 
         // Add delegate attributes as metadata before entry-specific conventions but after group conventions.
-        var attributes = handler.Method.GetCustomAttributes();
+        var attributes = entry.Method.GetCustomAttributes();
```

Comment:
Why is this using entry.Method and not methodInfo?

Reply:
```suggestion
        var attributes = methodInfo.GetCustomAttributes();
```